src/components/Invoice.tsx

```diff
@@ -8,108 +8,171 @@ import Button from '@mui/material/Button';
 import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 import IconButton from '@mui/material/IconButton';
+import { SubmitHandler, useForm, Controller } from 'react-hook-form';
+
+
+interface IPersonalData {
+
+  companyName: string | null;
+  city: string | null;
+  street: string | null;
+  postcode: string | null;
+  nip: number | null;
+  phoneNumber: number | null;
+  email: string | null,
+  bankAccount: string | null;
+}
+
+interface IInvoiceForm {
+  invoiceNumber: null,
+  dateFrom: Date | null;
+  dateTo: Date | null;
+  recipient: IPersonalData;
+  sender: IPersonalData
+}
 
 export default function Invoice() {
+  const { handleSubmit, register, control } = useForm<IInvoiceForm>({
+    defaultValues: {
+      invoiceNumber: null,
+      dateFrom: null,
+      dateTo: null,
+      recipient: {
+        companyName: null,
+        city: null,
+        street: null,
+        postcode: null,
+        nip: null,
+        phoneNumber: null,
+        email: null,
+        bankAccount: null,
+      },
+      sender: {
+        companyName: null,
+        city: null,
+        street: null,
+        postcode: null,
+        nip: null,
+        phoneNumber: null,
+        email: null,
+        bankAccount: null,
+      }
+    },
+  })
+
+  const onSubmit: SubmitHandler<IInvoiceForm> = (data) => console.log(data)
+
   return (
     <>
       <Grid container spacing={2} p={4} direction={'column'} mb={12}>
-        <Grid item container>
-          <Grid item xs={6} container>
-            <TextField fullWidth label="No." variant="standard" />
-            <Grid item container spacing={2} pt={2}>
-              <Grid item sm={6}>
-
-                <DatePicker />
-              </Grid>
-              <Grid item sm={6}>
-                <DatePicker />
+        <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
I don't think it works like now :D Looks like you are missing a button type on send button, so the submit event is never triggered.